server/game/GameActions/RevealCards.js

```diff
@@ -0,0 +1,107 @@
+const GameAction = require('./GameAction');
+const HandlerGameActionWrapper = require('./HandlerGameActionWrapper');
+const AcknowledgeRevealCardsPrompt = require('../gamesteps/AcknowledgeRevealCardsPrompt');
+
+class RevealCards extends GameAction {
+    constructor() {
+        super('revealCards');
+        this.defaultWhileRevealed = new HandlerGameActionWrapper({ handler: context => {
+                if(context.revealed.length > 0) {
+                    context.game.queueStep(new AcknowledgeRevealCardsPrompt(context.game, context.revealed, context.player))
+                }
+            }
+        });
+    }
+
+    canChangeGameState({ cards }) {
+        cards = Array.isArray(cards) ? cards : [cards];
+        return cards.length > 0 && cards.some(card => this.isInHiddenArea(card));
+    }
+
+    createEvent({ cards, player, whileRevealed, context }) {
+        const allPlayers = context.game.getPlayers();
+        const eventParams = {
+            player,
+            opponents: allPlayers.filter(p => p !== player),
+            cards: Array.isArray(cards) ? cards : [cards],
```

Comment:
I think I'd rather see us always pass it as an array, rather than try and figure out whether it's an array or not. `cards` already implies that it could be more than one.

Reply:
Adjusted in e2661e091c5a94eb156a653cf08385a597cf4e74. I've had to adjust `GameActions.Search` and `DeckSearchPrompt` to suit.